### PR TITLE
Fix support for pyspark jobs

### PR DESCRIPTION
**Pull request by Xeago**
Fixes https://github.com/spotify/spydra/issues/30 by @jrosskopf.

### CODE REVIEW

File: api/src/main/java/com/spotify/spydra/api/DataprocAPI.java [hunk 1]
```diff
@@ -86,7 +86,8 @@ public boolean deleteCluster(SpydraArgument arguments) throws IOException {
   public boolean submit(SpydraArgument arguments) throws IOException {
     boolean success = false;
     try {
-      success = gcloud.submit(arguments.getJobType(), arguments.getRegion(),
+      String pyFile = arguments.submit.pyFile.orElse(null);
```
Comment (steffengr) [inline api/src/main/java/com/spotify/spydra/api/DataprocAPI.java]: What is the reason to unpack the Optional here and then have a null check later on? Couldn't you simply pass the optional to gcloud.submit?

Comment (Xeago) [reply api/src/main/java/com/spotify/spydra/api/DataprocAPI.java]: Saw a message fly by my IDE, searched, found https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments. I prefer optionals too. :)

**Comment by Xeago** [comment]
@steffengr changes made, could you review again?

File: common/src/main/java/com/spotify/spydra/util/SpydraArgumentUtil.java [hunk 2]
```diff
@@ -172,6 +172,10 @@ public static void checkRequiredArguments(SpydraArgument arguments, boolean isOn
               + "defaultZones in configuration.");
         }
       }
+      if (arguments.getJobType() == SpydraArgument.JOB_TYPE_PYSPARK) {
+        arguments.submit.pyFile.orElseThrow(
+            () -> new IllegalArgumentException("pyspark jobs require the submit.pyfile ot be set"));
```
Comment (Cy6erBr4in) [inline common/src/main/java/com/spotify/spydra/util/SpydraArgumentUtil.java]: `s/ot be set/to be set`, also a space is missing between `submit.py` and `file`

**Pull request merged**